presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.h

```diff
@@ -99,6 +99,25 @@ class PartitionAndSerializeNode : public velox::core::PlanNode {
   const velox::core::PartitionFunctionSpecPtr partitionFunctionSpec_;
 };
 
+/// The state machine in the PartitionOperator consists of the following states:
+/// kUnknown - Represents an unknown state. Initial or unexpected state.
+///
+/// kProcessingInputElements - Indicates that input elements are being
+/// processed.
+///
+/// kProcessingNulls - Indicates that null elements are being processed.
+///
+/// kProcessingAny - Indicates that "any" elements are being processed.
+///
+/// kDone - Represents the end state of processing.
+enum ProcessingState {
```

Comment:
- Use enum class .
- Consider moving comments to individual values and dropping "Indicates that"
- Does this need to be defined in the header? Looks like it can go into .cpp file.

```
/// Process input rows.
kProcessingInputElements,

/// Process nulls.
kProcessingNulls,
```